stream_alert/alert_processor/outputs.py

```diff
@@ -199,6 +196,245 @@ def dispatch(self, **kwargs):
+
+    def _policy_verify(self, policy):
+        """Method to verify the existance of a escalation policy with the API
+
+        Args:
+            api_url (str): Base URL of the API
+            policy (str): Escalation policy to query about in the API
+            headers (dict): Headers used for API authentication
+
+        Returns:
+            dict: JSON object be used in the API call, containing the policy_id
+                  and escalation_policy_reference
+        """
+        policies_url = self._get_endpoint(self.base_url, self.POLICIES_ENDPOINT)
+
+        return self._item_verify(policies_url, policy, self.POLICIES_ENDPOINT,
+                                 'escalation_policy_reference')
+
+    def _service_verify(self, service):
+        """Method to verify the existance of a service with the API
+
+        Args:
+            api_url (str): Base URL of the API
+            service (str): Service to query about in the API
+            headers (dict): Headers used for API authentication
+
+        Returns:
+            dict: JSON object be used in the API call, containing the service_id
+                  and the service_reference
+        """
+        services_url = self._get_endpoint(self.base_url, self.SERVICES_ENDPOINT)
+
+        return self._item_verify(services_url, service, self.SERVICES_ENDPOINT, 'service_reference')
+
+    def _item_verify(self, item_url, item_str, item_key, item_type):
+        """Method to verify the existance of an item with the API
+
+        Args:
+            item_url (str): URL of the API Endpoint within the API to query
+            item_str (str): Service to query about in the API
+            item_key (str): Key to be extracted from search results
+            item_type (str): Type of item reference to be returned
+
+        Returns:
+            dict: JSON object be used in the API call, containing the item id
+                  and the item reference, or False if it fails
+        """
+        item_id = self._check_exists_get_id(item_str, item_url, item_key)
+        if not item_id:
+            LOGGER.info('%s not found in %s, %s', item_str, item_key, self.__service__)
+            return False
+
+        return {
+            'id': item_id,
+            'type': item_type
+        }
+
+    def dispatch(self, **kwargs):
+        """Send incident to Pagerduty Incidents API v2
+
+        Keyword Args:
+            **kwargs: consists of any combination of the following items:
+                descriptor (str): Service descriptor (ie: slack channel, pd integration)
+                rule_name (str): Name of the triggered rule
+                alert (dict): Alert relevant to the triggered rule
+                alert['context'] (dict): Provides user or escalation policy
+        """
+        creds = self._load_creds(kwargs['descriptor'])
+        if not creds:
+            return self._log_status(False)
+
+        # Preparing headers for API calls
+        headers = {
```

Comment:
suggestion: just set `self.headers` right here (or `self._headers` once the instance properties are made protected)